frontend/src/app/core/state/file-links/file-links.service.ts

```diff
@@ -70,16 +72,14 @@ export class FileLinksResourceService extends ResourceCollectionService<IFileLin
             return acc;
           }, seed));
         }),
-      )
-      .subscribe(
-        (fileLinkCollections) => {
+        map((fileLinkCollections) => {
           const storageId = idFromLink(fileLinkCollections.storage);
           const collectionKey = `${fileLinksSelfLink}?filters=[{"storage":{"operator":"=","values":["${storageId}"]}}]`;
           const collection = { _embedded: { elements: fileLinkCollections.fileLinks } } as IHALCollection<IFileLink>;
           insertCollectionIntoState(this.store, collection, collectionKey);
-        },
-        this.toastAndThrow.bind(this),
-      );
+        }),
+        catchError(this.toastAndThrow.bind(this)),
+      ) as Observable<void>;
```

Comment:
After a bit more thinking, I suggest a change:

- change return type to `Observable<IFileLink[]>`
- remove the error handling from here (remove `catchError`) and put the error handling (log and notify) into the caller
- and a `reduce` to the pipe, that aggregates the values of the observable stream.

Like it is now, this returned observable would emit value for every storage in the project of the work package. Here is the code suggestion:

```typescript
  updateCollectionsForWorkPackage(fileLinksSelfLink:string):Observable<IFileLink[]> {
    return this.http
      .get<IHALCollection<IFileLink>>(fileLinksSelfLink)
      .pipe(
        tap((collection) => insertCollectionIntoState(this.store, collection, fileLinksSelfLink)),
        switchMap((collection) => from(collection._embedded.elements)),
        groupBy(
          (fileLink) => fileLink._links.storage.href,
          (fileLink) => fileLink,
        ),
        mergeMap((group$) => {
          const seed = { storage: group$.key, fileLinks: [] as IFileLink[] };
          return group$.pipe(reduce((acc, fileLink) => {
            acc.fileLinks = [...acc.fileLinks, fileLink];
            return acc;
          }, seed));
        }),
        tap((fileLinkCollections) => {
          const storageId = idFromLink(fileLinkCollections.storage);
          const collectionKey = `${fileLinksSelfLink}?filters=[{"storage":{"operator":"=","values":["${storageId}"]}}]`;
          const collection = { _embedded: { elements: fileLinkCollections.fileLinks } } as IHALCollection<IFileLink>;
          insertCollectionIntoState(this.store, collection, collectionKey);
        }),
        reduce((acc, group) => acc.concat(group.fileLinks), [] as IFileLink[]),
      );
  }
```

Reply:
Applied, thanks!